java-checks/src/main/java/org/sonar/java/checks/BooleanMethodReturnCheck.java

```diff
@@ -53,6 +55,10 @@ private static boolean returnsBoolean(MethodTree methodTree) {
     return methodTree.returnType().symbolType().is("java.lang.Boolean");
   }
 
+  private static boolean hasUnknownAnnotation(SymbolMetadata symbolMetadata) {
```

Comment:
This looks like a good candidate for a new method on `SymbolMetadata` when the sprint is done